src/withForm/index.js

```diff
@@ -73,51 +80,82 @@ const processInput = ({ apiSchema, input }) => {
   return schema;
 };
 
-const generateField = (value, key) => {
-  if (!value.properties) {
-    if (value.items) {
-      return [];
-    }
-
-    return "";
-  }
+export const withForm = ({ input, formName, dataKey }) => {
+  const formData = formName ? `${formName}FormData` : `formData`;
+  const formErrors = formName ? `${formName}FormErrors` : `formErrors`;
+  const formUpdate = formName ? `${formName}FormUpdate` : `formUpdate`;
+  const schema = formName ? `${formName}Schema` : `schema`;
 
-  return _.mapObject(value.properties, (v, k) => generateField(v, k));
-};
-
-export const withForm = ({ input, formName, mergeKey }) => {
   return compose(
     setDisplayName(`withFormQewl(${formName})`),
     withStateHandlers(
-      ({ apiSchema, ...props }) => {
-        const processedSchema = processInput({ apiSchema, input });
-        const processedFormSchema = _.mapObject(
-          processedSchema.properties,
-          (v, k) => generateField(v, k)
-        );
-        return {
-          [`${formName}JSONSchema`]: processedSchema,
-          [`${formName}FormSchema`]: processedFormSchema,
-          [`${formName}FormData`]: omit(
-            {
-              ...processedFormSchema,
-              ...props[mergeKey]
-            },
-            ["__typename"]
-          )
-        };
-      },
+      ({ apiSchema, ...props }) => ({
+        [schema]: processInput({ apiSchema, input }),
+        [formData]: omit(
+          {
+            ...props[dataKey]
+          },
+          ["__typename"]
+        ),
+        [formErrors]: {
+          errors: null,
+          disabled: true
```

Comment:
Can we use a different name here, maybe `submitDisabled` or something?  In bigger forms it's going to beg the question 'what is disabled?`.  Right now I *think* this is for the submit button, a better name would self-document a lot better.